string/string.cc

```diff
@@ -1,12 +1,113 @@
 #include "string.h"
 
-String::String()
+#include <algorithm>
+#include <iterator>
+#include <memory>
+#include <string>
+
+
+const char* EMPTY = "";
+
+
+size_t strlen(const char* s)
```

Comment:
The library provides you with well-tested and super-optimized std::strlen. Do not reinvent the wheel.